assets/js/middleware/store-api-nonce.js

```diff
@@ -3,9 +3,18 @@
  */
 import apiFetch from '@wordpress/api-fetch';
 
-// @ts-ignore wcStoreApiNonce is window global
-// Cache for the initial nonce initialized from hydration.
-let nonce = wcStoreApiNonce || '';
+// Stores the current nonce for the middleware.
+let currentNonce = '';
+let currentTimestamp = 0;
+
+try {
+	const storedNonceValue = window.localStorage.getItem( 'storeApiNonce' );
+	const storedNonce = storedNonceValue ? JSON.parse( storedNonceValue ) : {};
```

Comment:
Any security concerns about storing the nonce value in localStorage? Would we be able to get away with just storing the nonce timestamp (and assume if that hasn't changed that the nonce in the request is okay)?

Reply:
@nerrad I don't think there is a security concern because the nonces are also injected into the page (so readable from JS), and they are to stop CSRF not used as secure tokens or anything. We need both here because the nonce injected on the page is out of date if you go back using the browser button. The new nonce needs to persist somewhere.

Reply:
Makes sense, in that case - this gets the 👍 from me (note, I have not tested though...so probably should get confirmed by someone else in testing).